### PR TITLE
[CN-337] Improve operator logging and use ginkgo cli

### PR DESCRIPTION
**Pull request by mtyazici**
Changes
- Added ginkgov2 CLI to Makefile. Now test logs are more informative.
- Improved logs for finalizer addition/deletion.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -143,11 +143,11 @@ test-it: manifests generate fmt vet ## Run tests.
 	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.8.3/hack/setup-envtest.sh
 	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); PHONE_HOME_ENABLED=$(PHONE_HOME_ENABLED) go test -v ./test/integration/... -coverprofile cover.out $(GO_TEST_FLAGS) -timeout 5m
 
-test-e2e: generate fmt vet ## Run end-to-end tests
-	USE_EXISTING_CLUSTER=true NAME_PREFIX=$(NAME_PREFIX) go test -v ./test/e2e -coverprofile cover.out -namespace "$(NAMESPACE)" -eventually-timeout 8m -timeout 30m -delete-timeout 8m $(GO_TEST_FLAGS)
```
Comment (hakanmemisoglu) [inline Makefile:147]: Can you use `-vv` instead of `-v`, and also add `--progress`?

File: Makefile [hunk 2]
```diff
@@ -143,11 +143,11 @@ test-it: manifests generate fmt vet ## Run tests.
...
+test-e2e: generate fmt vet ginkgo ## Run end-to-end tests
+	USE_EXISTING_CLUSTER=true NAME_PREFIX=$(NAME_PREFIX) $(GINKGO) -v -coverprofile cover.out ./test/e2e -- -namespace "$(NAMESPACE)" -eventually-timeout 8m -test.timeout 40m -delete-timeout 8m $(GO_TEST_FLAGS)
 	
-test-ph: generate fmt vet ## Run phone-home tests
```
Comment (hakanmemisoglu) [inline Makefile:149]: Same

File: controllers/hazelcast/hazelcast_resources.go [hunk 3]
```diff
@@ -34,11 +34,10 @@ const (
 )
 
 func (r *HazelcastReconciler) addFinalizer(ctx context.Context, h *hazelcastv1alpha1.Hazelcast, logger logr.Logger) error {
-	if !controllerutil.ContainsFinalizer(h, n.Finalizer) {
+	if !controllerutil.ContainsFinalizer(h, n.Finalizer) && h.GetDeletionTimestamp() == nil {
```
Comment (hakanmemisoglu) [inline controllers/hazelcast/hazelcast_resources.go:37]:
```suggestion
	if !controllerutil.ContainsFinalizer(h, n.Finalizer) && h.GetDeletionTimestamp().IsZero() {
```
Can you try this?

**Review by hakanmemisoglu — APPROVED** [review]
LGTM with a couple of comments

File: Makefile (hunk 1 above)
Comment (mtyazici) [reply Makefile:147]: I added them

File: controllers/hazelcast/hazelcast_resources.go (hunk 3 above)
Comment (mtyazici) [reply controllers/hazelcast/hazelcast_resources.go:37]: Does this add an improvement to the nil check? We actually do not care about the case where timestamp is zero. If it is not nil, then we know CR is going to be deleted.

File: go.sum [hunk 4]
```diff
@@ -139,6 +139,7 @@ github.com/go-openapi/spec v0.19.3/go.mod h1:FpwSN1ksY1eteniUU7X0N/BgJ7a4WvBFVA8
 github.com/go-openapi/swag v0.19.2/go.mod h1:POnQmlKehdgb5mhVOsnJFsivZCEZ/vjK9gh66Z9tfKk=
 github.com/go-openapi/swag v0.19.5/go.mod h1:POnQmlKehdgb5mhVOsnJFsivZCEZ/vjK9gh66Z9tfKk=
 github.com/go-stack/stack v1.8.0/go.mod h1:v0f6uXyyMGvRgIKkXu+yp6POWl0qKG85gN/melR3HDY=
+github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0 h1:p104kn46Q8WdvHunIJ9dAyjPVtrBPhSr3KT2yUst43I=
```
Comment (hakanmemisoglu) [inline go.sum:142]: @mtyazici Where is this coming from?

Comment (mtyazici) [reply go.sum:142]: It is a dependency of gingko v2 cli. I had to `go get` it manually.

File: controllers/hazelcast/hazelcast_resources.go (hunk 3 above)
Comment (hakanmemisoglu) [reply controllers/hazelcast/hazelcast_resources.go:37]: By looking at the implementation of the method, I would say it is safer. It checks `== nil` also. This is just a suggestion.

**Review by hasancelik — APPROVED** [review]

**Pull request merged**